partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt

```diff
@@ -278,8 +317,8 @@ internal class PartiQLPigVisitor(val customTypes: List<CustomType> = listOf(), p
 
     override fun visitDeleteCommand(ctx: PartiQLParser.DeleteCommandContext) = PartiqlAst.build {
         val from = visit(ctx.fromClauseSimple(), PartiqlAst.FromSource::class)
-        val where = visitOrNull(ctx.whereClause(), PartiqlAst.Expr::class)
-        val returning = visitOrNull(ctx.returningClause(), PartiqlAst.ReturningExpr::class)
+        val where = ctx.whereClause()?.let { visitWhereClause(it) }
+        val returning = ctx.returningClause()?.let { visitReturningClause(it) }
```

Comment:
The same reasoning as for `visit()` applies to `visitOrNull` here: 
- Remove the abstraction and deal with the null case in-line. 
- Avoid reflection for doing the cast. 
- Use an opportunity for a more specific method (`visitWhereClause`,   
  `visitReturningClause`).

Again, the overall verbosity is comparable if not better.

Reply:
> deal with the null case in-line

Agreed. The visitOrNull is coming from Java where the sugar isn't there